packages/core/test/utils_test.js

```diff
@@ -3827,5 +3828,41 @@ describe("utils", () => {
         }))
       );
     });
+    it("should infer correct anyOf schema based on data", () => {
```

Comment:
Let's also add a test case for oneOf. We could just parameterize this test in a loop, if that's easier, since the test cases are nearly exactly the same.

Reply:
Added almost identical test in `oneOf_test.js`